### PR TITLE
CLN: Remove PY2/PY3 flags from pandas.compat

### PR DESCRIPTION
**Pull request by jschendel**
- [X] xref #25725
- [X] tests passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by mroeschke** [comment]
Does this technically close #25725? Some of the functions in `compat/__init__.py` could still be cleaned up, but I think there's no more PY2 compat after this PR.

**Comment by gfyoung** [comment]
@mroeschke : I think it does. 🎉  I've updated the PR description accordingly.

@jschendel : Could you update the module docstring to indicate this no longer is compat for PY2?

**Comment by jschendel** [comment]
> Does this technically close #25725? 

There are still a few odds and ends left to address.  Will detail them in the main issue later on today.

> Could you update the module docstring to indicate this no longer is compat for PY2?

Sure, will get that done later today too.

**Review by mroeschke — APPROVED** [review]
LGTM

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -122,21 +120,12 @@ def __unicode__(self):
     def __str__(self):
         """
         Return a string representation for a particular Object
-
```
Comment (jreback) [inline pandas/core/dtypes/dtypes.py]: we might not even need to define str and unicode here and in base.py (future PR)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
merge away

**Comment by mroeschke** [comment]
Thanks @jschendel

**Pull request merged**